DistantWorlds2.ModLoader.Core/GameDataDefinitionPatching.cs

```diff
@@ -1164,6 +1376,32 @@ object ConvertToIdType(object id)
             int ConvertToInt(object id)
                 => ((IConvertible)id).ToInt32(NumberFormatInfo.InvariantInfo);
 
+            int GetIndex(List<T> defs, object id)
+            {
+                switch (defs)
+                {
+                    case IndexedList<T> indexed when id is int:
+                        return indexed.GetIndex(ConvertToInt(id));
+
+                    default:
+                        MethodInfo mi = defs.GetType().GetMethod("GetIndex");
+                        return (int)mi.Invoke(defs, new object[] { id });
+                }
+            }
+
+            bool ContainsId(List<T> defs, object id)
+            {
+                switch (defs)
+                {
+                    case IndexedList<T> indexed:
+                        return indexed.ContainsId(ConvertToInt(id));
+
+                    default:
+                        MethodInfo mi = defs.GetType().GetMethod("ContainsId");
+                        return (bool)mi.Invoke(defs, new object[] { id });
```

Comment:
how often is this called? for every item in a collection potentially, multiple times in multiple loops?